core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -2109,35 +1898,147 @@ func (o *evmTxStore) UpdateTxAttemptBroadcastBeforeBlockNum(ctx context.Context,
+	var cancel context.CancelFunc
+	ctx, cancel = o.stopCh.Ctx(ctx)
+	defer cancel()
+	var dbTxAttempts []DbEthTxAttempt
+	query := `
+		SELECT evm.tx_attempts.* FROM evm.tx_attempts
+		JOIN evm.txes ON evm.txes.ID = evm.tx_attempts.eth_tx_id
+		WHERE evm.tx_attempts.state = 'broadcast' AND evm.txes.state IN ('confirmed', 'confirmed_missing_receipt', 'fatal_error') AND evm.txes.evm_chain_id = $1 AND evm.txes.ID NOT IN (
+			SELECT DISTINCT evm.txes.ID FROM evm.txes
+			JOIN evm.tx_attempts ON evm.tx_attempts.eth_tx_id = evm.txes.ID
+			JOIN evm.receipts ON evm.receipts.tx_hash = evm.tx_attempts.hash
+			WHERE evm.txes.evm_chain_id = $1 AND evm.txes.state IN ('confirmed', 'confirmed_missing_receipt', 'fatal_error') AND evm.receipts.ID IS NOT NULL
+		)
+		ORDER BY evm.txes.nonce ASC, evm.tx_attempts.gas_price DESC, evm.tx_attempts.gas_tip_cap DESC
+	`
+	err = o.q.SelectContext(ctx, &dbTxAttempts, query, chainID.String())
+	attempts = dbEthTxAttemptsToEthTxAttempts(dbTxAttempts)
+	return attempts, err
+}
+
+// FindConfirmedTxesReceipts returns all confirmed transactions with receipt block nums older than or equal to the finalized block number
+func (o *evmTxStore) FindConfirmedTxesReceipts(ctx context.Context, finalizedBlockNum int64, chainID *big.Int) (receipts []*evmtypes.Receipt, err error) {
 	var cancel context.CancelFunc
 	ctx, cancel = o.stopCh.Ctx(ctx)
 	defer cancel()
 
+	var dbReceipts []Receipt
 	// note the receipts are partially loaded for performance reason
-	query := `SELECT evm.receipts.id, evm.receipts.tx_hash, evm.receipts.block_hash, evm.receipts.block_number FROM evm.receipts
+	query := `SELECT evm.receipts.tx_hash, evm.receipts.block_hash, evm.receipts.block_number FROM evm.receipts
 		INNER JOIN evm.tx_attempts ON evm.tx_attempts.hash = evm.receipts.tx_hash
 		INNER JOIN evm.txes ON evm.txes.id = evm.tx_attempts.eth_tx_id
 		WHERE evm.txes.state = 'confirmed' AND evm.receipts.block_number <= $1 AND evm.txes.evm_chain_id = $2`
-	err = o.q.SelectContext(ctx, &receipts, query, finalizedBlockNum, chainID.String())
+	err = o.q.SelectContext(ctx, &dbReceipts, query, finalizedBlockNum, chainID.String())
+	for _, dbReceipt := range dbReceipts {
+		receipts = append(receipts, &evmtypes.Receipt{
+			TxHash:      dbReceipt.TxHash,
+			BlockHash:   dbReceipt.BlockHash,
+			BlockNumber: big.NewInt(dbReceipt.BlockNumber),
+		})
+	}
 	return receipts, err
 }
 
-// Mark transactions corresponding to receipt IDs as finalized
-func (o *evmTxStore) UpdateTxStatesToFinalizedUsingReceiptIds(ctx context.Context, receiptIDs []int64, chainId *big.Int) error {
-	if len(receiptIDs) == 0 {
+// Mark transactions corresponding to attempt hashes as finalized
+func (o *evmTxStore) UpdateTxStatesToFinalizedUsingTxHashes(ctx context.Context, txHashes []common.Hash, chainID *big.Int) error {
+	if len(txHashes) == 0 {
 		return nil
 	}
+	txHashBytea := make([][]byte, len(txHashes))
```

Comment:
is `txHashBytea` a typo ?

Reply:
I think we're referring to the sql `bytea` type since this is in the tx store